app/src/main/java/fr/neamar/kiss/broadcast/UninstallShortcutHandler.java

```diff
@@ -0,0 +1,34 @@
+package fr.neamar.kiss.broadcast;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.util.Log;
+import fr.neamar.kiss.DataHandler;
+import fr.neamar.kiss.KissApplication;
+import fr.neamar.kiss.dataprovider.ShortcutProvider;
+import fr.neamar.kiss.pojo.ShortcutPojo;
+
+public class UninstallShortcutHandler extends BroadcastReceiver {
+
+    @Override
+    public void onReceive(Context context, Intent data) {
+
+        DataHandler dh = KissApplication.getDataHandler(context);
+        ShortcutProvider sp = dh.getShortcutProvider();
```

Comment:
if we go for the "shorcut handler can be null", don't forget to check for null here too ;)